### PR TITLE
Feat: added tests to the cloning and build methods

### PR DESCRIPTION
**Pull request by hmellin97**
Had to make some things public in the server file to make them accesible for the test class.

### CODE REVIEW

File: src/test/java/ContinuousIntegrationServerTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ContinuousIntegrationServerTest {
+
+    @Test
+    void cloneTheProject() {
```
Comment (Mathiponds) [inline src/test/java/ContinuousIntegrationServerTest.java]: We said last last that for Lab 2 we will make test method with at most one assert per function. There you can make a function `cloneTheProject_HelloWorldExists` and `cloneTheProject_GoodbyeWorldDontExists`

File: src/test/java/ContinuousIntegrationServerTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        ContinuousIntegrationServer.cloneTheProject(ContinuousIntegrationServer.gitHubRepoHTTPS);
+        File file1 = new File("assessmentDir/src/main/java/HelloWorld.java");
+        assertTrue(file1.exists());
+        File file2 = new File("assessment/src/main/java/GoodbyeWorld.java");
+        assertFalse(file2.exists());
+    }
+
+
+    @Test
+    void build() {
```
Comment (Mathiponds) [inline src/test/java/ContinuousIntegrationServerTest.java]: Can you make a test that assert `false`? I think the minimum for testing is having a true case and a false case to be sure that is doesn't always return the same value

**Review by Mathiponds — CHANGES_REQUESTED** [review]

File: src/test/java/ContinuousIntegrationServerTest.java (hunk 2 above)
Comment (hmellin97) [reply src/test/java/ContinuousIntegrationServerTest.java]: Yes, i'll fix that too!

File: src/test/java/ContinuousIntegrationServerTest.java (hunk 1 above)
Comment (hmellin97) [reply src/test/java/ContinuousIntegrationServerTest.java]: Yes, i will fix that!

File: src/main/java/ContinuousIntegrationServer.java [hunk 3]
```diff
@@ -18,8 +18,8 @@
 */
 public class ContinuousIntegrationServer extends AbstractHandler
 {
-    private static final String gitHubRepoHTTPS = "https://github.com/myhelmisaari/DD2480-Big-Brain-CI.git";
-    private static final String assessmentRepo = "assessmentDir/";
+    public static final String gitHubRepoHTTPS = "https://github.com/myhelmisaari/DD2480-Big-Brain-CI.git";
+    public static final String assessmentRepo = "assessmentDir/";
```
Comment (axelkennedal) [inline src/main/java/ContinuousIntegrationServer.java]: I think should be `protected`, not public.

File: src/main/java/ContinuousIntegrationServer.java [hunk 4]
```diff
@@ -63,7 +63,7 @@ public static void main(String[] args) throws Exception
      * Clones the assessment branch from the GitHub repository given as argument.
      * @param gitHubHTTPS the https of the repository we want to clone
      */
-    private static void cloneTheProject(String gitHubHTTPS) {
+    public static void cloneTheProject(String gitHubHTTPS) {
```
Comment (axelkennedal) [inline src/main/java/ContinuousIntegrationServer.java]:
```suggestion
    protected static void cloneTheProject(String gitHubHTTPS) {
```

File: src/main/java/ContinuousIntegrationServer.java [hunk 5]
```diff
@@ -80,7 +80,7 @@ private static void cloneTheProject(String gitHubHTTPS) {
      * directory given as argument
      * @param assessmentRepo the directory that contains the project we want to build
      */
-    private static void build(String assessmentRepo){
+    public static void build(String assessmentRepo){
```
Comment (axelkennedal) [inline src/main/java/ContinuousIntegrationServer.java]:
```suggestion
    protected static void build(String assessmentRepo){
```

File: src/test/java/ContinuousIntegrationServerTest.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    void cloneTheProject_HelloWorldExists() {
+        ContinuousIntegrationServer.cloneTheProject(ContinuousIntegrationServer.gitHubRepoHTTPS);
+        File file = new File("assessmentDir/src/main/java/HelloWorld.java");
+        assertTrue(file.exists());
+    }
+
+    @Test
+    void cloneTheProject_GoodByeWorldDoesNotExist() {
+        ContinuousIntegrationServer.cloneTheProject(ContinuousIntegrationServer.gitHubRepoHTTPS);
+        File file = new File("assessmentDir/src/main/java/GoodByeWorld.java");
```
Comment (axelkennedal) [inline src/test/java/ContinuousIntegrationServerTest.java:19]: hahah I like this test

**Review by axelkennedal — CHANGES_REQUESTED** [review]
just change the access modifiers

File: src/main/java/ContinuousIntegrationServer.java (hunk 3 above)
Comment (Mathiponds) [reply src/main/java/ContinuousIntegrationServer.java]: It should be private but for the tests need to have access to those variables

**Review by Mathiponds — APPROVED** [review]

**Review by axelkennedal — APPROVED** [review]

**Pull request merged**